lib/model/query/analytics.js

```diff
@@ -661,6 +688,36 @@ const countOwnerOnlyDatasets = () => ({ oneFirst }) => oneFirst(sql`
   SELECT COUNT(1) FROM datasets where "ownerOnly" = true;
 `);
 
+const countDatasetsWithGeometry = () => ({ oneFirst }) => oneFirst(sql`
+  SELECT COUNT(DISTINCT "datasetId") FROM ds_properties WHERE name = 'geometry';
+`);
+
+const countEntitiesWithGeometry = () => ({ all }) => all(sql`
+SELECT
+  COUNT(*) AS total,
+  COUNT(CASE WHEN e."createdAt" >= ${_cutoffDate}
+    THEN 1
+    ELSE null
+  END) AS recent,
+  ds."projectId",
+  ds.id as "datasetId"
+FROM entities AS e
+  JOIN entity_defs AS ed ON ed."entityId" = e.id AND ed.current = true
+  JOIN datasets AS ds ON ds.id = e."datasetId"
+WHERE ed.data ? 'geometry'
+  AND e."deletedAt" IS NULL
+  AND ds."publishedAt" IS NOT NULL
```

Comment:
Seeing this condition on `publishedAt` made me wonder, should we also filter on `publishedAt` in the other new queries here? Like in `countFormsCreateUpdateEntitiesFromRepeats()` or `countDatasetsWithGeometry()`? Published datasets, published properties. Probably it doesn't matter, but I started to wonder why only this metric filters on it.

Reply:
Sounds good about not changing things now. :+1:

> `countFormsCreateUpdateEntitiesFromRepeats` - could exclude draft forms?

Ah my suggestion probably wouldn't have an effect in this case after all, since you're already filtering for published forms by joining on `forms."currentDefId"`.